modules/account-lib/src/coin/algo/seedEncoding.ts

```diff
@@ -0,0 +1,88 @@
+import * as base32 from 'hi-base32';
+import sha512 from 'js-sha512';
+import { Seed } from './ifaces';
+
+const SEED_BYTES_LENGTH = 32;
+const ALGORAND_SEED_BYTE_LENGTH = 36;
+const ALGORAND_CHECKSUM_BYTE_LENGTH = 4;
+const ALGORAND_SEED_LENGTH = 58;
+
+const MALFORMED_SEED_ERROR = new Error('seed seems to be malformed');
```

Comment:
nit: create the error where it gets thrown, otherwise the stack trace won't be very helpful.